source/concepts/flakes.md

```diff
@@ -40,53 +44,74 @@ Nix handles flakes differently than regular {term}`Nix file`s in the following w
 
   It means full reproducibility of a Nix expression, and, by extension, the resulting build instructions by default, but also prohibits parameterisation of results by consumers.
 
+(flakes-controversy)=
 ## Why are flakes controversial?
 
-{ref}`Flakes <flakes>` were originally proposed in [RFC 49](https://github.com/NixOS/rfcs/pull/49), and have been in development since 2019.
+[Flakes](flakes-definition) were inspired by [Shea Levy's NixCon 2018 talk](https://www.youtube.com/watch?v=DHOLjsyXPtM), formally proposed in [RFC 49](https://github.com/NixOS/rfcs/pull/49), and have been in development since 2019.
 Nix introduced the implementation as its first [experimental feature] in 2021.
 
-[experimental feature]: https://nix.dev/manual/nix/2.18/contributing/experimental-features.html
-
-The subject is considered controversial among Nix users and developers in terms of design, development processes, and community governance.
+The subject is considered controversial among Nix users and developers in terms of design, implementation quality, and decision making process.
 In particular:
-- The RFC was closed without conclusion, and some design and implementation issues are not yet resolved.
-  Examples include the notion of a global [flake registry], the [impossibility of parameterising flakes](https://github.com/NixOS/nix/issues/2861), and the [new command line interface and flakes being closely tied to each other](https://discourse.nixos.org/t/2023-03-06-nix-team-meeting-minutes-38/26056#cli-stabilisation-announcement-draft-4).
-- The original implementation introduced [regressions](https://discourse.nixos.org/t/nix-2-4-and-what-s-next/16257) in the [Nix 2.4 release](https://nix.dev/manual/nix/2.18/release-notes/rl-2.4.html), breaking some stable functionality without a [major version](https://semver.org/) increment.
-- New Nix users were and still are encouraged by various individuals to adopt flakes despite there being no concrete plan or timeline for stabilisation.
-
-[flake registry]: https://nix.dev/manual/nix/2.18/command-ref/new-cli/nix3-registry.html
+- The RFC was closed without conclusion, and some fundamental issues are not yet resolved.
+  For example:
+  - The notion of a [global flake registry](https://github.com/NixOS/flake-registry) saw [substantial criticism](https://github.com/NixOS/rfcs/pull/49#issuecomment-635635333) that was never addressed.
+    Additionally, local registry names in Nix expressions [introduce mutable system state](https://github.com/NixOS/nix/issues/7422) and are thus no improvement over [channels](https://nix.dev/manual/nix/2.18/command-ref/nix-channel) in that regard.
```

Comment:
It's comparable much more to `NIX_PATH` rather than channels (which also shouldn't be confused with `nix-channel`). I suggest:

> Additionally, local registry names in Nix expressions [introduce mutable system state](https://github.com/NixOS/nix/issues/7422) and are thus in that regard no improvement over [`NIX_PATH`](https://nix.dev/manual/nix/2.18/command-ref/env-common.html#env-NIX_PATH), often managed using [`nix-channel`](https://nix.dev/manual/nix/2.18/command-ref/nix-channel).

Reply:
Hm, no, registry names are not as arbitrary as lookup paths. The analog are really channels IMO, because those are system-wide, persistent names for source references - both *could* be used to populate search paths. I don't like the "channels file system stuff" page in the Nix manual all that much because it's not very informative conceptually, this is why I linked `nix-channel`, which incorporates the other one anyway.

Reply:
> Hm, no, registry names are not as arbitrary as lookup paths.

Agreed, makes sense.

> The analog are really channels

Not channels in general though, really just channels as managed by `nix-channel`. The general concept of channels is really just a tested source distribution, which even flake registries depend on. So I think this should be:

```suggestion
    Additionally, local registry names in Nix expressions [introduce mutable system state](https://github.com/NixOS/nix/issues/7422) and are thus no improvement over [`nix-channel`](https://nix.dev/manual/nix/2.18/command-ref/nix-channel) in that regard.
```

Or

```suggestion
    Additionally, local registry names in Nix expressions [introduce mutable system state](https://github.com/NixOS/nix/issues/7422) and are thus no improvement over channels as managed by [`nix-channel`](https://nix.dev/manual/nix/2.18/command-ref/nix-channel) in that regard.
```

Reply:
```suggestion
    Additionally, local registry names in Nix expressions [introduce mutable system state](https://github.com/NixOS/nix/issues/7422) and are thus, in that regard, no improvement over [channels](https://nix.dev/manual/nix/2.18/command-ref/nix-channel) – but as opposed to channels, the source references of [registry entries can be pinned](https://nix.dev/manual/nix/2.19/command-ref/new-cli/nix3-registry-pin).
```